pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminApiHealthCheckTest.java

```diff
@@ -55,16 +61,91 @@ public void cleanup() throws Exception {
 
     @Test
     public void testHealthCheckup() throws Exception {
-        admin.brokers().healthcheck();
+        final int times = 30;
+        new Thread(() -> {
+            for (int i = 0; i < times; i++) {
+                try {
+                    admin.brokers().healthcheck();
+                    Thread.sleep(20);
+                } catch (PulsarAdminException | InterruptedException e) {
+                    Assert.fail();
```

Comment:
This won't usually work well as assertion errors should escape from the test method.
You can use a CompletareFuture and complete it exceptionally

Reply:
Fixed